geopandas/geodataframe.py

```diff
@@ -392,6 +524,8 @@ def merge(self, *args, **kwargs):
         if isinstance(result, DataFrame) and geo_col in result:
             result.__class__ = GeoDataFrame
             result.crs = self.crs
+            g = vectorized.VectorizedGeometry(result[geo_col])
+            result[geo_col] = list(g)
```

Comment:
This entire function was written assuming that we placed pointers into the dataframe directly.  This approach is currently broken.